mozilla/blurts-server
# MNTOR-1385 Basic backend flow for Premium featureset

# References: 
Jira: [MNTOR-1385](https://mozilla-hub.atlassian.net/browse/MNTOR-1385)

# Description

Add the backend elements necessary to run the basic Premium subscription flow.

Some important notes:
1. This creates two new routes, `/oauth/premium/upgrade` and `/oauth/premium/confirmed`, which drive the flow. They are not real views but are responsible for redirects and validation.
2. The whole feature is behind a feature flag. Set `FXA_SUBSCRIPTION_ENABLED=1` in your `.env` to enable.

# How to test

1. Run the app with the feature flag disabled. You should see no changes on the monitor dashboard. If you navigate to either of the two routes listed above you should see a 404 error.
2. Enable the feature flag. Log in to monitor with an FxA account that does not have a subscription. You should see a link in the top right of the dashboard that says "Upgrade to Premium". Click the link and follow the subplat flow. When you return to monitor you should be returned to the dashboard and the link should be replaced with a string that says "Premium subscriber".


# TODO
- [ ] Add a comment to the README explaining how the interaction between the staging subplat URLs and local development works
- [ ] I am not really sure how to test this as I don't see a unit test suite that exists for the `src/` code.

# Checklist (Definition of Done)
